lib/gui/modules/child-writer.ts

```diff
@@ -302,3 +320,43 @@ ipc.connectTo(IPC_SERVER_ID, () => {
 		ipc.of[IPC_SERVER_ID].emit('ready', {});
 	});
 });
+
+async function saveFileBeforeFlash(
+	imagePath: string,
+	saveUrlImageTo: string,
+	onProgress: (state: ProgressState) => void,
+	onFail: (
+		destination: sdk.sourceDestination.SourceDestination,
+		error: Error,
+	) => void,
+) {
+	const urlImage = new Http({ url: imagePath, avoidRandomAccess: true });
+	const source = await urlImage.getInnerSource();
```

Comment:
Are we sure we want to save the decompressed image ?

Reply:
Good call, how do we deal with decompressing the image though? Should we download with a normal axios call then call the `decompressThenFlash` method on that file or flash without decompressing first?

Reply:
Good call, I'll change that to pipeToDesinations then decompress afterwards, so we can also check the compressed size in case it's a compressed image